color/src/dynamic.rs

```diff
@@ -347,6 +351,43 @@ impl DynamicColor {
         other: Self,
         cs: ColorSpaceTag,
         direction: HueDirection,
+    ) -> Interpolator {
+        self.interpolate_with_alpha(other, cs, direction, InterpolationAlphaSpace::Premultiplied)
+    }
+
+    /// Interpolate two colors.
+    ///
+    /// The colors are interpolated linearly from `self` to `other` in the color space given by
+    /// `cs`. When interpolating in a cylindrical color space, the hue can be interpolated in
+    /// multiple ways. The [`direction`](`HueDirection`) parameter controls the way in which the
+    /// hue is interpolated.
+    ///
+    /// The interpolation proceeds according to [CSS Color Module Level 4 § 12][css-sec].
+    ///
+    /// This method does a bunch of precomputation, resulting in an [`Interpolator`] object that
+    /// can be evaluated at various `t` values.
+    ///
+    /// [css-sec]: https://www.w3.org/TR/css-color-4/#interpolation
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use color::{AlphaColor, InterpolationAlphaSpace, ColorSpaceTag, DynamicColor, HueDirection, Srgb};
+    ///
+    /// let start = DynamicColor::from_alpha_color(AlphaColor::<Srgb>::new([1., 0., 0., 1.]));
+    /// let end = DynamicColor::from_alpha_color(AlphaColor::<Srgb>::new([0., 1., 0., 1.]));
+    ///
+    /// let interp = start.interpolate_with_alpha(end, ColorSpaceTag::Hsl, HueDirection::Increasing, InterpolationAlphaSpace::Premultiplied);
+    /// let mid = interp.eval(0.5);
+    /// assert_eq!(mid.cs, ColorSpaceTag::Hsl);
+    /// assert!((mid.components[0] - 60.).abs() < 0.01);
+    /// ```
+    pub fn interpolate_with_alpha(
+        self,
+        other: Self,
+        cs: ColorSpaceTag,
+        direction: HueDirection,
+        interpolation_alpha_space: InterpolationAlphaSpace,
```

Comment:
Perhaps like `unpremultiplied_gradient` this should be `unpremultiplied_interpolate` and not take the enum. That prevents having to make the `InterpolationAlphaSpace` public.

(Though if we're renaming, `interpolate_unpremultiplied` sounds nicer, so perhaps it should also be `gradient_unpremultiplied`.)